src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -532,7 +532,8 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                 return Task.FromResult(actualItem);
             }
 
-            pyItem.OnNewAnalysis += (o, e) => OnPythonEntryNewAnalysis(pyItem);
+            pyItem.NewAnalysis += OnProjectEntryNewAnalysis;
+            pyItem.Disposed += (s, e) => ((IPythonProjectEntry)s).NewAnalysis -= OnProjectEntryNewAnalysis;
```

Comment:
I'd say that this is actually a responsibility of `ProjectEntry` itself - it should clear its own event handlers on dispose.

Reply:
OK

Reply:
Actually on RemovedFromProject I guess

Reply:
Replace RemovedFromProject  by Dispose?